app/store/index.js

```diff
@@ -1,13 +1,123 @@
+
+import {ViewTypes} from 'app/constants';
 import appReducer from 'app/reducers';
-import configureServiceStore from 'mattermost-redux/store';
+
+import {transformSet} from './utils';
 
 function getAppReducer() {
     return require('../../app/reducers'); // eslint-disable-line global-require
 }
 
-export default function configureStore(preloadedState) {
-    return configureServiceStore(preloadedState, appReducer, getAppReducer);
+const usersSetTransform = [
+    'profilesInChannel',
+    'profilesNotInChannel',
+    'profilesInTeam'
+];
+
+const teamSetTransform = [
+    'membersInTeam'
+];
+
+const setTransforms = [
+    ...usersSetTransform,
+    ...teamSetTransform
+];
+
+export default function configureStore(initialState) {
+    const viewsBlackListFilter = createBlacklistFilter(
+        'views',
+        ['login']
+    );
+
+    const setTransformer = createTransform(
+        (inboundState, key) => {
+            if (key === 'entities') {
+                const state = {...inboundState};
+                for (const prop in state) {
+                    if (state.hasOwnProperty(prop)) {
+                        state[prop] = transformSet(state[prop], setTransforms);
+                    }
+                }
+
+                return state;
+            }
+
+            return inboundState;
+        },
+        (outboundState, key) => {
+            if (key === 'entities') {
+                const state = {...outboundState};
+                for (const prop in state) {
+                    if (state.hasOwnProperty(prop)) {
+                        state[prop] = transformSet(state[prop], setTransforms, false);
+                    }
+                }
+
+                return state;
+            }
+
+            return outboundState;
+        }
+    );
+
+    const offlineOptions = {
+        persist: (store, options) => {
+            const persistor = persistStore(store, {storage: AsyncStorage, ...options}, () => {
+                store.dispatch({
+                    type: Constants.STORE_REHYDRATION_COMPLETE,
+                    complete: true
+                });
+            });
+
+            let purging = false;
+
+            // check to see if the logout request was successful
+            store.subscribe(() => {
+                const state = store.getState();
+                if (state.requests.users.logout.status === RequestStatus.SUCCESS && !purging) {
+                    purging = true;
+
+                    store.dispatch(batchActions([
+                        {
+                            type: Constants.OFFLINE_STORE_RESET,
+                            data: initialState
+                        },
+                        {
+                            type: ViewTypes.SERVER_URL_CHANGED,
+                            serverUrl: state.views.selectServer.serverUrl
+                        }
+                    ]));
+
+                    persistor.purge();
```

Comment:
Well technically yes, but in this case no. Purge will clear only the state that is stored in the `AsyncStorage` and leave the application state alone. When another action is fired the state is persisted again. Since we fire the modal close action the server url actually does get persisted. But it should be moved because it does make more sense. I'll move it and submit.